src/readblock.c

```diff
@@ -0,0 +1,299 @@
+/****************************************************************************************
+
+    FFIO - Flexible format I/O
+
+    FFIO provides functionality for reading proj style instruction strings written
+    in a less strict format than usual:
+
+    *  Whitespace is generally allowed everywhere
+    *  Comments can be written inline, '#' style
+    *  ... or as free format blocks
+
+    The overall mission of FFIO is to facilitate communications of geodetic
+    parameters and test material in a format that is highly human readable,
+    and provides ample room for comment, documentation, and test material.
+
+    See the PROJ ".gie" test suites for examples of supported formatting.
+
+****************************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+
+#include <string.h>
+#include <ctype.h>
+
+#include <math.h>
+#include <errno.h>
```

Comment:
I don't follow the order of headers here.  Is there some reason for the 3 groups and unusual order?  Why not just alphabetize them and remove the groupings

Reply:
I have never met the idiom of alphabetic inclusion order before, but it's a good idea for cases where the order otherwise doesn't matter, and I will reorganize here, and try to remember following the rule in new code

Reply:
I usually follow the Python recommendations for this. Standard library headers first,  external headers and finally local headers. Since we don't have any external dependencies it's just stdlib then local headers, preferably separated by a space. But since the PROJ.4 codebase is a big pile of disorder with regards to code style it hardly matters much. There's no such thing as consistency here :)

Reply:
... but at least we could strive for *orderly* disorder